osc-domain/src/main/java/org/osc/core/broker/model/entities/virtualization/SecurityGroupInterface.java

```diff
@@ -25,182 +28,189 @@
+	private VirtualSystem virtualSystem;
+
+	@ManyToMany(fetch = FetchType.LAZY)
+	@JoinTable(name = "SECURITY_GROUP_INTERFACE_POLICY",
+			joinColumns = @JoinColumn(name = "sgi_fk", referencedColumnName = "id"),
+			inverseJoinColumns = @JoinColumn(name = "policy_fk", referencedColumnName = "id"))
+	private Set<Policy> policies = new HashSet<>();
+
+	/**
+	 * The tag is assumed to be in the format "SOMESTRING" "-" "LONG VALUE".
+	 * isc-456 for example.
+	 */
+	@Column(name = "tag", nullable = true)
+	private String tag;
+
+	@ManyToOne(fetch = FetchType.EAGER)
+	@JoinColumn(name = "security_group_fk", nullable = true, foreignKey = @ForeignKey(name = "FK_SECURITY_GROUP"))
+	private SecurityGroup securityGroup;
+
+	@Column(name = "user_configurable", columnDefinition = "bit default 0")
+	private boolean isUserConfigurable;
+
+	@Column(name = "mgr_interface_id")
+	private String mgrSecurityGroupIntefaceId;
+
+	@Column(name = "mgr_security_group_id")
+	private String mgrSecurityGroupId;
+
+	@Column(name = "failure_policy_type", nullable = false)
+	@Enumerated(EnumType.STRING)
+	private FailurePolicyType failurePolicyType = FailurePolicyType.NA;
+
+	@Column(name = "chain_order", columnDefinition = "bigint default 0", nullable = false)
+	private Long order = 0L;
+
+	/**
+	 * Represents the identifier of the inspection hook created in the SDN controller
+	 * when it supports port groups.
+	 */
+	@Column(name = "network_elem_id")
+	private String networkElementId;
+
+	public SecurityGroupInterface(VirtualSystem virtualSystem, Set<Policy> policies, String tag,
+			FailurePolicyType failurePolicyType, Long order) {
+		super();
+		this.virtualSystem = virtualSystem;
+		this.policies = policies;
+		this.tag = tag;
+		this.isUserConfigurable = true;
+		this.failurePolicyType = failurePolicyType;
+		this.order = order;
+	}
+
+	public SecurityGroupInterface() {
+		super();
+	}
+
+	public Set<Policy> getPolicies() {
+		return this.policies;
+	}
+
+	public void setPolicies(Set<Policy> policies) {
+		this.policies = policies;
+	}
+
+	public String getName() {
+		return this.name;
+	}
+
+	public void setName(String name) {
+		this.name = name;
+	}
+
+	public String getTag() {
+		return this.tag;
+	}
 
-    @ManyToOne(fetch = FetchType.EAGER)
-    @JoinColumn(name = "virtual_system_fk", nullable = false,
-    foreignKey = @ForeignKey(name = "FK_SG_VIRTUAL_SYSTEM"))
-    private VirtualSystem virtualSystem;
+	public void setTag(String tag) {
+		this.tag = tag;
+	}
+
+	public Long getTagValue() {
+		return this.tag == null ? null : Long.valueOf(this.tag.substring(this.tag.indexOf("-") + 1));
+	}
 
-    @ManyToOne(fetch = FetchType.EAGER)
-    @JoinColumn(name = "policy_fk", foreignKey = @ForeignKey(name = "FK_SGI_POLICY"))
-    private Policy policy;
+	public Set<String> getMgrPolicyIds() {
+		return this.policies.stream().map(Policy::getMgrPolicyId).collect(Collectors.toSet());
```

Comment:
guard against null pointers? I beleive policies can be null if policy mapping is not supported